src/server/dfly_main.cc

```diff
@@ -177,37 +181,50 @@ struct VersionMonitor {
   }
 
  private:
-  void RunTask();
+  void RunTask(SSL_CTX* ssl_ctx);
 };
 
-void VersionMonitor::RunTask() {
+void VersionMonitor::Run(ProactorPool* proactor_pool) {
+  // Don't run in dev environment - i.e. where we don't build with
+  // real version number
+  if (!GetFlag(FLAGS_version_check) || kGitTag[0] != 'v')
+    return;
+
+  SSL_CTX* ssl_ctx = TlsClient::CreateSslContext();
+  if (!ssl_ctx) {
+    VLOG(1) << "Remote version - failed to create SSL context - cannot run version monitoring";
+    return;
+  }
+
+  version_fiber_ =
+      proactor_pool->GetNextProactor()->LaunchFiber([ssl_ctx, this] { RunTask(ssl_ctx); });
+}
+
+void VersionMonitor::RunTask(SSL_CTX* ssl_ctx) {
   const auto loop_sleep_time = std::chrono::hours(24);  // every 24 hours
 
   const std::string host_name = "version.dragonflydb.io";
   const std::string_view port = "443";
   const std::string resource = "/v1";
-  const std::string version_header = std::string("DragonflyDB/") + kGitTag;
+  string_view current_version(kGitTag);
 
-  SSL_CTX* context = TlsClient::CreateSslContext();
-  if (!context) {
-    VLOG(1) << "Remote version - failed to create SSL context - cannot run version monitoring";
-    return;
-  }
+  current_version.remove_prefix(1);
+  const std::string version_header = absl::StrCat("DragonflyDB/", current_version);
 
   ProactorBase* my_pb = ProactorBase::me();
   while (true) {
     const std::optional<std::string> remote_version =
-        GetRemoteVersion(my_pb, context, host_name, port, resource, version_header);
+        GetRemoteVersion(my_pb, ssl_ctx, host_name, port, resource, version_header);
     if (remote_version) {
       const std::string rv = remote_version.value();
-      if (rv != kGitTag) {
-        LOG_FIRST_N(INFO, 1) << "Your current version '" << kGitTag
+      if (rv != current_version) {
+        LOG_FIRST_N(INFO, 1) << "Your current version '" << current_version
                              << "' is not the latest version. A newer version '" << rv
                              << "' is now available. Please consider an update.";
       }
     }
     if (monitor_ver_done_.WaitFor(loop_sleep_time)) {
-      TlsClient::FreeContext(context);
+      TlsClient::FreeContext(ssl_ctx);
```

Comment:
since this is a class, and you moved this outside this function, maybe we can move this to  a dtor? will make the ssl free context more consistent